### ISSUE TITLE
Hologram with that name already exists after destroying it

### ISSUE THREAD

**Issue by viveleroi** [DecentSoftware-eu/DecentHolograms#58]
### Just making sure

- [X] I made sure, this bug is not already known or reported.
- [X] I believe, that this bug is fixable.

### Reproduction

1. Install DecentHolograms 2.4.2

I'm using the API to create a hologram:

`Hologram hologram = DHAPI.createHologram(identifier, location, lines)`

When server operators use a command to reload our configs, I destroy all holograms I created:

`hologram.destroy();`

I have even tried to also remove the hologram by name:

`DHAPI.removeHologram(this.hologram.getName());`

Next the createHologram logic above runs again since we're re-initializing everything with updated config data.

DH errors with: `Hologram with that name already exists! (testcrate_x-2_y79_z-15)`

Despite having not saved the file, destroyed the instance, and removed it from the API, it still exists in memory.
        

### Solution

Fix code so I can destroy and then re-make the hologram.

### Server Version

git-Paper-52 (MC: 1.19) (Implementing API version 1.19-R0.1-SNAPSHOT) (Git: 6ea73e7)

### Client Version

1.19

### Plugin Version

2.4.2

### Log

https://paste.helpch.at/eyekiyehan.md

**Comment by d0by1** [comment]
Hello, could you please show me the full code?

**Comment by viveleroi** [comment]
When a crate instance is loaded/created we create holograms:

https://github.com/darkhelmet-gaming/StellarCrates/blob/main/stellarcrates/src/main/java/network/darkhelmet/stellarcrates/services/crates/CrateInstance.java#L91

The hologram is created: https://github.com/darkhelmet-gaming/StellarCrates/blob/main/stellarcrates/src/main/java/network/darkhelmet/stellarcrates/services/holograms/providers/DecentHologramsProvider.java#L35

When crate instances are unloaded, we destroy all holograms: https://github.com/darkhelmet-gaming/StellarCrates/blob/main/stellarcrates/src/main/java/network/darkhelmet/stellarcrates/services/crates/CrateInstance.java#L122

Hologram is destroyed: https://github.com/darkhelmet-gaming/StellarCrates/blob/main/stellarcrates/src/main/java/network/darkhelmet/stellarcrates/services/holograms/providers/DecentHologram.java#L45

**Comment by viveleroi** [comment]
Yes, I have checked multiple times and have used debugging to ensure it's called on every hologram I've created.

**Comment by viveleroi** [comment]
Well, I don't see any difference in the code being used outside of yours being a test and mine not. Except you're calling `hologram.delete();` on plugin disable, which I don't because I was told that was only for saved holograms. Either way none of this is a post-shutdown issue.

The process for my plugin is:

1. On server boot, load crate configs
2. create a hologram for each location
3. player runs reload configs cmd
4. I call destroy on all holograms
5. I rerun the hologram create logic

I threw some println in there to be sure my code is running. No errors in the console.

```
[10:02:10 INFO]: [StellarCrates] [STDOUT] creating hologram testcrate_x-2_y79_z-15 // server start
[10:02:23 INFO]: viveleroi issued server command: /crates reloadconfig
[10:02:23 INFO]: [StellarCrates] [STDOUT] destroying hologram testcrate_x-2_y79_z-15
[10:02:23 INFO]: [StellarCrates] [STDOUT] creating hologram testcrate_x-2_y79_z-15
... java.lang.IllegalArgumentException: Hologram with that name already exists! (testcrate_x-2_y79_z-15)
```

Also, just to be thorough I tried 2.5.1 that was just released and have the same issue.

**Comment by d0by1** [comment]
Well, you kinda have to at least destroy the holograms when your plugin disables. Although the file does not exist, the hologram is still in the memory. DecentHolograms API saves it in cache even after your plugin disables. So definitely destroy holograms in your onDisable method. 

I don't really know why this is happening to you. Have you tried getting the hologram instead of creating a new one? If the hologram already exists, you wouldn't need to create it. It should be removed after calling the destroy method, I even checked the code of DH multiple times, but couldn't you just use the instance that supposedly already exists?

Also, I noticed you are using the forEach method: `holograms.forEach(CrateHologram::destroy);`. And that's completely fine but maybe try using a normal for loop instead? I think I has some issues with the forEach method before but I'm not sure.

**Comment by viveleroi** [comment]
Ok this was definitely my issue, although it's probably not good long-term for destroy to leave holograms in memory in a way we need to also call `delete` later.

While my logic to destroy and create the holograms was all good, the entire chain was running *twice* due to an error in my configuration reload logic. So it was effectively creating a hologram and then making it again, thus triggering the error.

Thanks for your time, this was difficult to track down.
